config/initializers/cookie_rotator.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# From https://github.com/rails/rails/pull/44967
+Rails.application.config.after_initialize do
+  Rails.application.config.action_dispatch.cookies_rotations.tap do |cookies|
+    authenticated_encrypted_cookie_salt = Rails.application.config.action_dispatch.authenticated_encrypted_cookie_salt
+    signed_cookie_salt = Rails.application.config.action_dispatch.signed_cookie_salt
```

Comment:
In multisite, the salt varies for each site. The salts are [set in the `env`](https://github.com/discourse/rails_multisite/blob/main/lib/rails_multisite/cookie_salt.rb#L13) very early in the middleware stack. So I guess ideally we need to do all this calculation at runtime, after that middleware has run 🤔

Reply:
Indeed we need to find a way to make this working with multisite :thinking: Well, I’m going to investigate some more and conduct some tests :sweat_smile: